common/rpc/grpc.go

```diff
@@ -178,23 +180,12 @@ func NewServiceErrorInterceptor(
 		resp, err := handler(ctx, req)
 		if err != nil {
 			addHeadersForResourceExhausted(ctx, logger, err)
-		}
-		return resp, serviceerror.ToStatus(err).Err()
-	}
 
-}
+			if _, ok := err.(*serviceerrors.ShardOwnershipLost); ok {
+				err = serviceerror.NewUnavailable(shardUnavailableErrorMessage)
+			}
+		}
```

Comment:
we cannot do it here because this interceptor is used by history service as well. Frontend service rely on this error to be returned from history service so it could do a proper retry (and find the new owner from the error details). We want to mask this only at frontend before it returns to external caller. 

Can we do this in MaskInternalErrorsInterceptor (which I assume is only used by frontend)

Reply:
Discussed offlline. Will revive "FrontendErrorInterceptor"

Reply:
I believe that it shouldn't be added at all. Apparently it is handled somewhere else.

Reply:
also will add serviceerror.DataLoss